### ISSUE TITLE
Proposal: Routing on Area Features

### ISSUE THREAD

**Issue by kevinkreiser** [valhalla/valhalla#3570]
The tag of highway=platform and its support in valhalla came up in a discussion where the comments eventually focused on tagging conventions and the ability for the router to traverse area features. From that discussion an idea for a proposal was suggested: https://github.com/valhalla/valhalla/discussions/3565#discussioncomment-2333589

The basic gist of the proposal is:

1. parse area features like parks etc that would seem like good candidates for which to provide routing
    1. this may mean checking if the area already intersects traversable ways (ie its already mapped)
    1. it may also mean checking if any routable ways are near by 
1. create an internal network for the polygon using either its boundary or perhaps the medial axis
1. if the boundary is connected to the road network use it otherwise find connection points like we do with transit imports

This issue is just a place holder until more time exists to think about how to implement something like this. It should also be noted that this is more of a problem for OSM than any commercial dataset, but it could be argued that such a feature would enable routing (especially indoors or within other enclosed areas) where it wouldnt be possible even with commercial datasets.

**Comment by slvlirnoff** [comment]
Yes it's a super interesting features, especially for complexes areas, like square with fountains, path, places or big railway stations.

One naïve approach could also be to find all intersection of the area with street network, and connect these two by two if visible (i.e. not stepping outside the polygon).
This would work rather well on convex polygon but fails for more complex shapes or multi-polygon with holes.

A look at how OpenTripPlanner is handling these cases could be interesting, I know they have a good areaVisibility feature now that works quite well for transit. I'm not sure what OSRM, GraphHopper, etc are doing in this area?

Also OSM tend to rather ban / go away from additional street segment within areas and encourage implementation of this feature. Adding additional data in OSM for sake of routing is usually not the right approach.

Relevant issue with lot's of information
https://github.com/osmlab/osm-planning/issues/17

Also
https://github.com/valhalla/valhalla/issues/42

Also quick solution as preprocessing step (but I've never tried it)
https://github.com/PlazaRoute/plazaroute

**Comment by jthiard** [comment]
There is a recent PR for OSRM, using a visibility graph + dijkstra (to reduce the number of generated edges).

https://github.com/Project-OSRM/osrm-backend/pull/7161

**Comment by kevinkreiser** [comment]
@jthiard thanks for the heads up, like dennis was saying about osrm, we wanted to do something like that for a long time but never found the time! very nice work from that contributor over there!

**Comment by nilsnolde** [comment]
Wow, did they really implement to respect building footprints in the visibility graph? That's crazy:) I figured that'd be insanely slow, then I might retract my comment here, https://github.com/valhalla/valhalla/issues/5184#issuecomment-2792265545.

**Comment by jthiard** [comment]
I suppose building footprints inside the area are respected only if they are inner rings of the pedestrian area ?
Which suppose tagging the area as a ["multipolygon" relation](https://wiki.openstreetmap.org/wiki/Relation:multipolygon).

**Comment by nilsnolde** [comment]
Ah yeah that makes sense!

**Comment by chrstnbwnkl** [comment]
This can now be closed after #6195 has landed!
